x-pack/plugins/maps/public/classes/sources/es_search_source/es_search_source.tsx

```diff
@@ -725,6 +727,29 @@ export class ESSearchSource extends AbstractESSource implements ITiledSingleLaye
         : urlTemplate,
     };
   }
+
+  async getTimesliceMaskFieldName(): Promise<string | null> {
+    const indexPattern = await this.getIndexPattern();
+    return indexPattern.timeFieldName ? indexPattern.timeFieldName : null;
+  }
+
+  canMaskTimeslice(prevMeta: DataMeta, timeslice?: Timeslice): boolean {
```

Comment:
Is there a way we can move the majority of this check  to the layer-level, and have sources indicate more something like 'canPrefetchData`?


- metadata for mvt-sources about 'completeness' will be loaded quite differently, and probably not be part of the layer-state, at least in how we're thinking about mvt-metadata right now: cf. https://github.com/elastic/kibana/pull/94811
- already we need to introduce some "hacks" for a layer backfilling the metadata https://github.com/elastic/kibana/pull/96791/files#diff-e38a737f9e12768aa864d38710d91b7b809413a9c55d7d299e5595ef86198300R340
- "masking" seems a very style-centric operation, which I'm not sure we want to introduce into the source. In other words, can we remove _all_ knowledge about timeslices from sources? (e.g. it's basically the same as `isTimeAware()`,  so maybe just  change `getTimesliceMaskFieldName` to `getTimeFieldName`).
 
^  fwiw - it's hard for me know to image how https://github.com/elastic/kibana/pull/94811 and this PR will connect cleanly.